### PR TITLE
Complete Kubernetes resources from all namespaces only if namespace is unset for the context

### LINKED ISSUE #240: Provide a way to opt out of `--all-namespaces` option in kubectl
### Command name

kubectl

### Completion context

```shell
kubectl get all **<TAB>
```

### What happens now?

All resources are completed with `--all-namespaces` option enabled.

### What do you expect to complete?

Any way to disable `--all-namespaces` option is desired since one might not have a permission to list resources in all namespaces, which requires `ClusterRole`.

### PR DESCRIPTION
**Pull request by chitoku-k**
Closes #240

### CODE REVIEW

File: src/core/parser.zsh [hunk 1]
```diff
@@ -162,7 +162,7 @@ _fzf_complete_parse_option() {
         fi
     done
 
-    if [[ -z $result ]]; then
+    if [[ $#result = 0 ]]; then
```
Comment (hexium310) [inline src/core/parser.zsh:165]: Why is this changed? In `_fzf_complete_kubectl_default_arguments` `-z` is used, so I wonder at those differences.

Comment (chitoku-k) [reply src/core/parser.zsh:165]: This specific change is just for consistency with the rest of the similar implementations in this file. However, the change in `_fzf_complete_parse_option_arguments` is to fix a bug where it previously returned `1` when the given types were `'argument'` and the provided command was like `-n ""` whose result should have been `('')` that was considered empty.

**Review by hexium310 — APPROVED** [review]
